tasks/elasticsearch-plugins.yml

```diff
@@ -46,18 +46,20 @@
   environment:
     CONF_DIR: "{{ conf_dir }}"
     ES_INCLUDE: "{{ instance_default_file }}"
+    ES_ENV_FILE: "{{ instance_default_file }}"
 
 - name: Install elasticsearch plugins
-  command: "{{es_home}}/bin/elasticsearch-plugin install {{ item.plugin }} --batch --silent"
+  command: "{{es_home}}/bin/elasticsearch-plugin install {{ item.url | default(item.plugin) }} --batch --silent"
   register: plugin_installed
   failed_when: "'ERROR' in plugin_installed.stdout"
   changed_when: plugin_installed.rc == 0
   with_items: "{{ es_plugins }}"
-  when: item.plugin in plugins_to_install
+  when: item.plugin is defined and item.plugin in plugins_to_install
   notify: restart elasticsearch
   environment:
     CONF_DIR: "{{ conf_dir }}"
     ES_INCLUDE: "{{ instance_default_file }}"
+    ES_ENV_FILE: "{{ instance_default_file }}"
```

Comment:
@orphaner why do you need the `ES_ENV_FILE`? `CONF_DIR` is not enough?

Reply:
Trying to identify what the supported way forward here. It seems both are supported i.e.

```
# The default env file is defined at building/packaging time.
# For a distribution package, the value is "/etc/default/elasticsearch".
ES_ENV_FILE="/etc/default/elasticsearch"
# If an include is specified with the ES_INCLUDE environment variable, use it
if [ -n "$ES_INCLUDE" ]; then
    ES_ENV_FILE="$ES_INCLUDE"
fi
```
however, id like to use the one that will be supported moving forward.